### PR TITLE
Enable CI testing with dev infrastructure.

### PR DESCRIPTION
**Pull request by benpatterson**
Hi again. I had some additional cycles the last couple of days to grok and set up some additional infrastructure.

This also addresses #13 

This change does a couple of things.
- Adds testing to the Travis CI infrastructure. (This repo has one test, and that is exercised.)
- Reports coverage on that testing. (We could wire this up to [coveralls](https://coveralls.io/) or [codecov](https://codecov.io/) if you want a badge; that shouldn't be too hard but someone with github admin privileges would need to enable it on one of those tools)
- Sets a separate settings file for development.  On that:
  - To run on a separate CI system, it's less complicated to have run on sqlite (in memory) and the setup is easier. This could change to postgres later if that is really what you'd prefer.
  - This settings file can be used for local development for folks by using sqlite, which can be a lot easier for people who want to start contributing.
  - I realize this settings file is not DRY. A better approach could be to use a base settings file that is called upon by the other settings files (dev, production, etc), but I didn't wire that up in this pull request because that may end up resulting in changes for openshift and make this PR turn into something larger than it really needs to be. So that could be done later.
- Added some doc on running tests locally

I saw a volunteer form online somewhere; I will fill that out as an introduction at the least.

### CODE REVIEW

**Comment by pwnage101** [comment]
Great to see the possibility of running tests in sqlite!  I wonder if this codebase uses any postgres-specific database features.

>  Reports coverage on that testing. (We could wire this up to coveralls or codecov if you want a badge; that shouldn't be too hard but someone with github admin privileges would need to enable it on one of those tools)

So, without coveralls or codecov, we only see coverage if we click through to the travis ci console output?  Sounds easy enough to wire up one of those and add a badge.

**Comment by benpatterson** [comment]
Yes, I am being optimistic with sqlite, but it is more performant for running tests since it'll default to in-memory when being used for unit tests. (Besides, installing postgres is more overhead and I was trying to get started quickly.) This could also potentially be test.py instead of dev.py, wherein dev is for local development and is postgres. (Updating the tests to use postgres is not terrible; I was just opting for the faster system for tests.)

And you're right on the coverage. Those tools can also enforce a coverage % for the changes themselves, if that is worthwhile.

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[run]
+data_file = .coverage
+omit =
+    checkin2015/settings*
+    *wsgi.py
+    *migrations*
+    *admin.py
+    *static*
+    *templates*
+
```
Comment (pwnage101) [inline .coveragerc:10]: shouldn't this also omit `*test*` since that code can't cover itself?

Comment (benpatterson) [inline .coveragerc:10]: my philosophy has been to cover test code, in part so you get the fixtures that you want to be testing, and in part to ensure you're running all tests. But I'm not firm on that position...I can add a filter for it.

**Comment by pwnage101** [comment]
I wanted to test the dev settings module, so I created the local sqlite database using:

```
DJANGO_SETTINGS_MODULE=checkin2015.dev
export DJANGO_SETTINGS_MODULE
python manage.py migrate
python manage.py runserver 0.0.0.0:8000
```

Added some dates to the months table and the survey becomes open.  Will have to dig in later to see where this will break, but it seems that some basic funtionality works.  I'm confident merging this now.

**Comment by thecristen** [comment]
Not sure how much it matters, but it looks like `dev.py` was copied from an earlier `settings.py` - at the very least, the email settings in `dev.py` are outdated. Wondering how many of these settings could/ should be shared between the two files...

**Comment by pwnage101** [comment]
Restating Ben, indeed the settings files are not DRY, but the immediate concern here is that changing settings.py to import from some sort of common configuration file without some minimal testing might cause unexpected behavior or failed deployment.  For the time being, the code's pathway to the production settings module is completely unchanged.  Should probably also add a reminder in #13 to "make settings files DRY".

**Pull request merged**